### PR TITLE
feat: PolicyStudio - add prerequisiteMessage for SPG

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

https://gravitee.atlassian.net/browse/APIM-5391

**Description**

image

**Additional context**

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-schematics
```
npm install @gravitee/ui-schematics@12.20.0-prerequisitemessage-0633ed0
```
```
yarn add @gravitee/ui-schematics@12.20.0-prerequisitemessage-0633ed0
```

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-particles-angular
```
npm install @gravitee/ui-particles-angular@12.20.0-prerequisitemessage-0633ed0
```
```
yarn add @gravitee/ui-particles-angular@12.20.0-prerequisitemessage-0633ed0
```

---

### CODE REVIEW

File: projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.harness.ts [hunk 1]
```diff
@@ -44,6 +45,13 @@ export class GioPolicyStudioDetailsPhaseStepHarness extends ComponentHarness {
     return (await (await this.locatorForOptional(DivHarness.with({ selector: '.info__description' }))())?.getText()) ?? undefined;
   }
 
+  public async getInfoMessage(): Promise<string | undefined> {
+    const tooltip = await this.locatorForOptional(MatTooltipHarness.with({ selector: '.info__head__info' }))();
+    // Fixme: await tooltip.getTooltipText(); returns empty string. when it's in a condition. I haven't figured out how to get over it.
```
Comment (ytvnr) [inline projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.harness.ts:50]:
```suggestion
    // FIXME: await tooltip.getTooltipText(); returns empty string. when it's in a condition. I haven't figured out how to get over it.
```
not sure it's properly scanned in camel case ?

File: projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.component.html [hunk 2]
```diff
@@ -25,6 +25,9 @@
     <span *ngIf="genericPolicy?.type === 'SHARED_POLICY_GROUP'" class="gio-badge-neutral" matTooltip="Shared Policy Group">
       <mat-icon svgIcon="gio:users"></mat-icon>
     </span>
+    @if (getPrerequisiteMessage) {
+      <mat-icon class="info__head__info" svgIcon="gio:info" [matTooltip]="getPrerequisiteMessage"></mat-icon>
```
Comment (ytvnr) [inline projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.component.html:29]: will it be good to have a tooltip with a potential big text ?

**Review by ytvnr — APPROVED** [review]

File: projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.component.html (hunk 2 above)
Comment (ThibaudAV) [reply projects/ui-policy-studio-angular/src/lib/components/flow-details-phase-step/gio-ps-flow-details-phase-step.component.html:29]: I don't really have anything better :/ and that's what figma wants . I'll add it in the description part of the catalog too.

**Pull request merged**